src/AgreementManager.ts

```diff
@@ -485,11 +488,9 @@ class AgreementManager {
     //both canonical chain and future queue
     public isBlockInChain(block: BlockStruct): boolean {
         const agreement = this.getAgreementByBlock(block);
-
-        return agreement
-            ? EvmUtils.encodeBlock(agreement.block) ==
-                  EvmUtils.encodeBlock(block)
-            : false;
+        return (
+            (agreement || false) && this.areBlocksEqual(agreement.block, block)
+        );
```

Comment:
Not sure why "|| false" is here

Reply:
Need to convert undefined to boolean.
Without it we get
`Type 'boolean | undefined' is not assignable to type 'boolean'. Type 'undefined' is not assignable to type 'boolean'.ts(2322)`